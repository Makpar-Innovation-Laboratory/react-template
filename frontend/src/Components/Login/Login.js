import React, {useContext} from 'react'
import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button'
import axios from 'axios'
import Auth from '../../Utility/Auth'
import {Context} from '../../App'
import {useNavigate} from 'react-router-dom'

export default function Login () {
  const navigate = useNavigate()
  const context = useContext(Context)
  const handleOnSubmit = (e) => {
    e.preventDefault()

    const data = { username: e.target[0].value, password: e.target[1].value }
    console.log('We submitted! : ')
    axios
      .post('https://api-laboratory-dev.makpar-innovation.com/token', data, {
        headers: {
          'content-type': 'application/json'
        }
      })
      .then((res) => {
        // console.log('Success: ', res.data)
        Auth.authenticateUser(res.data.AuthenticationResult.IdToken)
        context.setAuth(true)
        navigate('/')
      })
  }
  
  return (
    <div className='container w-25'>
      <Form onSubmit={handleOnSubmit} >
        <Form.Group className='mb-3' controlId='formBasicEmail'>
          <Form.Label>Username</Form.Label>
          <Form.Control type='text' placeholder='Enter Username' name='username' />
          <Form.Text className='text-muted'>just work</Form.Text>
        </Form.Group>

        <Form.Group className='mb-3' controlId='formBasicPassword'>
          <Form.Label>Password</Form.Label>
          <Form.Control type='password' placeholder='Password' name='password' />
        </Form.Group>
        <Form.Group className='mb-3' controlId='formBasicCheckbox'>
          <Form.Check type='checkbox' label='Check me out' />
        </Form.Group>
        <Button variant='primary' type='submit' >
          Submit
        </Button>
      </Form>
    </div>
  )
}
