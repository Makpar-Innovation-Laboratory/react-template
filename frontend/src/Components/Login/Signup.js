import React, {useState} from "react";

import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button'

function SignUpForm () {
    
    const handleOnSubmit = (e) => {
        e.preventDefault();
        const data = { email: e.target[0].value, username: e.target[1].value, dev: e.target[3].value }
        axios
        .post('api-innolab-dev.makpar-innovation.net/register', data, {
            headers: {
            'content-type': 'application/json'
            }
        })
        .then((res) => {
            console.log('Success: ', res.data)
            navigate('/')
        })
    }
    
    return (
    <div className='container w-25'>
        <div className="d-flex justify-content-center mt-5">
          <h3 className="color-primary" data-testid='title'>Sign Up</h3>
        </div>
        <div>
            <Form onSubmit={handleOnSubmit} data-testid='LoginForm'>
                <Form.Group className='mb-3' data-testid='formBasicEmail'>
                <Form.Label>Username</Form.Label>
                <Form.Control type='text' placeholder='Enter Username' name='username' data-testid='username' id='username'/>
                </Form.Group>

                <Form.Group className='mb-3' controlId='formBasicPassword'>
                <Form.Label>Password</Form.Label>
                <Form.Control type='password' placeholder='Password' name='password' data-testid='password'/>
                </Form.Group>
                <Form.Group>
                    <Form.Check 
                        type='checkbox'
                        name='developer-checkbox'
                    />
                </Form.Group>
                <Button variant='primary' type='submit' title="Login button" data-testid='button' id='button'>
                    Sign Up
                </Button>
            </Form>
            <a href="/Login" >Already Have an account?</a>
        </div>
    </div>
  );
};

export default SignUpForm;