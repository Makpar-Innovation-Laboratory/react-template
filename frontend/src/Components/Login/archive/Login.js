import React, {useContext, useState} from 'react'
import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button'
import axios from 'axios'
import Auth from '../../../Utility/Auth'
import {Context} from '../../../App'
import {useNavigate} from 'react-router-dom'

/**
 * @component
 * @description description goes here
 * @returns {}
 */
export default function Login () {
  const navigate = useNavigate()
  const context = useContext(Context)
  const [loading, setLoading] = useState(false)
  /**
   * @description function for handling API authentication 
   * 
   * @param {*} e  - form submission containing username/password combo for authentication
   */
  const handleOnSubmit = (e) => {
    e.preventDefault()
    setLoading(true)
    const data = { username: e.target[0].value, password: e.target[1].value}
    console.log(data)
    console.log('We submitted! : ')
    axios
      .post('https://api-innolab-dev.makpar-innovation.net/token', data, {
        headers: {
          'content-type': 'application/json'
        }
      })
      .then((res) => {
        // console.log('Success: ', res.data)
        Auth.authenticateUser(res.data.AuthenticationResult.IdToken)
        context.setAuth(true)
        context.updateUser(e.target[0].value)
        context.updateUserid(123)
        setLoading(false)
        navigate('/')
      })
  }
  
  return (
    <div className='container w-25'>
      <Particles options={particlesConfig} id="tsparticles"/>
      <main className="box">
        <Form onSubmit={handleOnSubmit}>
          <div className="d-flex justify-content-center mt-5">
            <h3 className="color-primary" data-testid='title'>Login</h3>
          </div>
          <Form.Group className='mb-3' controlId='formBasicEmail'>
            <Form.Label>Username</Form.Label>
            <Form.Control type='text' placeholder='Enter Username' name='username' data-testid='username' id='username'/>
          </Form.Group>

          <Form.Group className='mb-3' controlId='formBasicPassword'>
            <Form.Label>Password</Form.Label>
            <Form.Control type='password' placeholder='Password' name='password' data-testid='password'/>
          </Form.Group>
          {loading === false?(
            <Button variant='primary' type='submit' title="Login button" data-testid='button' id='button'>
              Login
            </Button>):(
              <PacmanLoader color="#0000ff" loading={true} size={20} css={override}/>
            )}
          {loading === false?(
            <a href="/Signup" >Don't have an account?</a>):(
              <h3>loading</h3>
            )} 
          </Form>
        </main>
    </div>
  )
}
