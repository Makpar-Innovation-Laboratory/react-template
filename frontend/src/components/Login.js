import React from 'react';


export default function Login () {
  // Connected
  return (
    <div className='container w-25'>
      Login Page
    </div>
  )
}
