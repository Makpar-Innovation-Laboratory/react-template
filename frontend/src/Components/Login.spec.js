import React from "react";
import { isolateComponent } from "isolate-react";
import Login from "./Login";
// import { shallow, mount } from "enzyme";

describe("JSX Tests :: <Header /> Rendering", () => {

  test("Sanity Check Snapshot Rendering", () => {
    const wrapper = isolateComponent(<Login />);

    expect(wrapper.debug()).toMatchSnapshot();
  });

});
