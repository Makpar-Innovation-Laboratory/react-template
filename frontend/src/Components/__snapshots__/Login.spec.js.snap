// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSX Tests :: <Header /> Rendering Sanity Check Snapshot Rendering 1`] = `
"{\\"nodeType\\":\\"html\\",\\"type\\":\\"div\\",\\"name\\":\\"div\\",\\"props\\":{\\"className\\":\\"container w-25\\",\\"children\\":\\"Login Page\\"},\\"key\\":\\"null\\"}
 -> {\\"nodeType\\":\\"string\\",\\"type\\":\\"Login Page\\",\\"props\\":{},\\"name\\":\\"\\",\\"key\\":\\"undefined\\"}"
`;
